### ISSUE TITLE
Add more languages

### ISSUE THREAD

**Issue by ovidius72** [SmiteshP/nvim-gps#4]
Very nice plugin. Are you planning to add more languages?.
 I'd like to see typescript ad typescript react (.tsx)

**Comment by SmiteshP** [comment]
Hey! I am willing to add more languages to the list :)
To begin with I stuck to languages I am proficient in and/or have some knowledge about the language. Please feel free to make a PR for your favorite languages.
I will soon add a contributions section to readme to explain how the tree-sitter queries work!

**Comment by Chaitanyabsprip** [comment]
I would like support for dart

**Comment by SmiteshP** [comment]
Hey @ovidius72! I just added some basic queries for typescript.

**Comment by ovidius72** [comment]
@SmiteshP Thank you. 
The next weekend I would like to have a look at the Treesitter API and see in if I can help in some way.

**Comment by SmiteshP** [comment]
That would be great!!

**Comment by SmiteshP** [comment]
Hello @ovidius72 and @danielebra, I recently added support for jsx and tsx files too! As far as I understand they are pretty similar to javascript and typescript respectively so the same treesitter queries should work with them too. 
And thanks to @Lazytangent's contributions, the queries for javascript and typescript have improved! 😃

**Comment by ovidius72** [comment]
@SmiteshP Thank you. I'm sorry I didn't have the time to check it out.
I'm just playing with it right now. I configured it with lualine as for you example.
In tsx files it show the number "1" in place of methods.
In .ts file it shows nothing for me.
![Notification_Center](https://user-images.githubusercontent.com/3592928/132132054-1a840948-a212-4e28-a2ef-ad4319a67651.png)

typescript file. Inside an arrow function.
![Notification_Center](https://user-images.githubusercontent.com/3592928/132132084-8e8709c3-99aa-4e7f-8e94-22e200d004b4.png)

the result of `lua print(require('nvim-gps').is_available()` give me `nil`

Thanks.

**Comment by ovidius72** [comment]
Here is the lualine section where i show up the gps module info.

```
  sections = {
    lualine_a = {{'mode', format=function(mode) return mode:sub(1,1) end }},
    lualine_b = {
      {'filename', { file_status = true, path = 1  }}
    },
    lualine_c = {
      {
        'diagnostics',
        sources = { 'coc' },
        -- displays diagnostics from defined severity
        sections = {'error', 'warn', 'info', 'hint'},
        -- all colors are in format #rrggbb
        color_error = '#eb6f92', -- changes diagnostic's error foreground color
        color_warn = '#f6c177', -- changes diagnostic's warn foreground color
        color_hint = '#9ccfd8', -- Changes diagnostic's info foreground color
        color_info = '#31748f', -- Changes diagnostic's hint foreground color
        symbols = { error = ' ', warn = ' ', info = ' ', hint = ' '}
      },
      { gps.get_location, condition = gps.is_available },
      {
        "g:coc_status"
      }
    },

```

**Comment by SmiteshP** [comment]
Hey @ovidius72, could you open a new issue regarding this and share some code snippets so that I can reproduce this error.

**Comment by Hubro** [comment]
I added support for YANG! :tada: https://github.com/SmiteshP/nvim-gps/pull/24

<sub>Although I suspect I am the only person in the world who uses neovim + nvim-gps and codes in YANG...</sub>

**Comment by Chaitanyabsprip** [comment]
> Hey @Chaitanyabsprip! After looking into it, unfortunately I don't think it will be possible to support dart in this plugin. Because unlike other languages, the treesitter for dart creates function signature and function body as sibling nodes. So there will be no single node that can be captured as `scope-root`. Until this behavior changes its impossible to write queries for this :( This issue here discusses this behavior [UserNobody14/tree-sitter-dart#12](https://github.com/UserNobody14/tree-sitter-dart/issues/12)

Hey, I just came across the FlutterOutline Command in the flutter-tools by akinsho. you could look into that how he's implemented that to add support for dart in your plugin

**Comment by Mo8it** [comment]
@SmiteshP Thank you for this awesome package!

Is it possible to add support for Julia?

I did take a look at how to add a language, but it is not that intuitive.

Thanks :)

**Comment by SmiteshP** [comment]
Hey @Chaitanyabsprip @Mo8it and @Suyashtnt !
Could you checkout my other plugin https://github.com/SmiteshP/nvim-navic ?
It is similar to nvim-gps but makes use of LSP instead of treesitter. So it won't need any special attention for every language, it will work as long as there is a language server for that language.

**Comment by Suyashtnt** [comment]
> Hey @Chaitanyabsprip @Mo8it and @Suyashtnt !
> 
> Could you checkout my other plugin https://github.com/SmiteshP/nvim-navic ?
> 
> It is similar to nvim-gps but makes use of LSP instead of treesitter. So it won't need any special attention for every language, it will work as long as there is a language server for that language.

Is there a way to integrate this into something like winbar.nvim?

**Comment by SmiteshP** [comment]
Yep! If you are using feline.nvim just add the component to feline's winbar setup function. Or you can also use the native neovim way.

**Comment by Suyashtnt** [comment]
> Yep! If you are using feline.nvim just add the component to feline's winbar setup function. Or you can also use the native neovim way.

I'm talking about the plugin called winbar.nvim, is there a way to use this in a way that resembles it's look
